docs/robots/husky/tutorials_husky.mdx

```diff
@@ -4,7 +4,1535 @@ sidebar_position: 4
+    <img
+      src="/img/robot_images/husky_images/rqtgraph.png"
+      width="800"
+    />
+    <figcaption>rqt_graph snapshot</figcaption>
+  </figure>
+</center>
+
+The highlighted node and arrow show the topic that you are publishing to the simulated Husky. This Husky then goes on to update the gazebo virtual environment,
+which takes care of movement of the joints (wheels) and the physics of the robot.
+The `rqt_graph` command is very handy to use, when you are unsure who is publishing to what in ROS.
+Once you figure out what topic you are interested in, you can see the content of the topic using `rostopic echo`.
+
+#### Using tf
+
+In ROS, tf is a special topic that keeps track of coordinate frames, and how they relate to each other.
+The simulated Husky starts at (0,0,0) in the world coordinate frame. When the Husky moves, its own coordinate
+frame changes. Each wheel has a coordinate frame that tracks how it is rotating, and where it is. Generally,
+anything on the robot that is not fixed in space, will have a tf describing it.
+In the rqt_graph section above, you can see that the `/tf` topic is published to and subscribed from by many
+different nodes.
+
+One intuitive way to see how the `tf` topic is structured for a robot is to use the `view_frames` tool
+provided by ROS. Open a terminal window and run:
+
+```bash
+rosrun tf view_frames
+```
+
+Wait for this to complete, and then run:
+
+```
+evince frames.pdf
+```
+
+This will bring up something similar to the following image.
+
+<center>
+  <figure>
+    <img
+      src="/img/robot_images/husky_images/tfframes.png"
+      width="800"
+    />
+    <figcaption>tf frames snapshot</figcaption>
+  </figure>
+</center>
+
+Here we can see that all four wheel are referenced to the `base_link`. We also see that the `odom` topic
+is driving the reference of the whole robot. This means that if you write to the `odom` topic,
+such as when you publish to the `/cmd_vel` topic, then the whole robot will move.
+
+### Keeping Husky Updated
+
+:::note
+If you are upgrading your Husky from an older version of ROS, please refer to
+our upgrade instructions [here](https://clearpathrobotics.com/assets/guides/kinetic/kinetic-to-melodic/index.html)
+and [here](https://clearpathrobotics.com/assets/guides/melodic/melodic-to-noetic/index.html).
+:::
+
+#### Getting New Packages
+
+Jackal is always being improved, both its own software and the many community ROS packages upon which it depends!
+You can use the `apt` package management system to receive new versions all software running on the platform.
+
+Each Jackal leaves the factory already configured to pull packages from http://packages.ros.org as well as
+http://packages.clearpathrobotics.com. To update your package and download new package versions, simply run:
+
+```bash
+sudo apt-get update
+sudo apt-get dist-upgrade
+```
+
+#### MCU Firmware Update
+
+Customer updates of Husky firmware are not supported at this time. Contact [Support](#support) if
```

Comment:
Should Puma firmware updates be mentioned here to prevent confusion?

/docs/robots/husky/maintenance/#maintenance_husky_motor_controller_firmware

Reply:
I've added the link.